### DISCUSSION TITLE
Try + Catch User Auth Flutter Dart

### DISCUSSION THREAD

**Discussion by rodnnnney** [pocketbase/pocketbase#4867] [Q&A] [answered]
Hey guys! I'm pretty new to pocket base and want to know how do you catch errors when dealing with user authentication in Pocket base? 

I want to be able to catch errors when users try input their info sign in and display a UI element corresponding to the error they made.

I tried parsing the error like JSON
Getting the status code of the thrown error

This is what I am working with below: 
```
Future<void> signIn(String email, String password) async {
    try {
      final authData =
          await pb.collection('users').authWithPassword(email, password);
      setAuthData = authData;
      notifyListeners();
    } catch (error) {
      print('Error during sign in: $error}');
    }
  }

```

When I tried get the response type of the error `error.response` and map it statusCode and handle it from there. Seems like [] operator can't be applied on the error object. However, since the console is printing this, I think it should be possible, the message is telling you what is wrong. 

`Error during sign in: ClientException: {url: ---/auth-with-password, isAbort: false, statusCode: 400, response: {code: 400, message: Something went wrong while processing your request., data: {identity: {code: validation_required, message: Cannot be blank.}, password: {code: validation_required, message: Cannot be blank.}}}, originalError: null}`

All I want is the data{identity:...} and password:{code...}

I couldn't find anything about this in the documentation. Any help or ideas is greatly appreciated.

**Comment by ganigeorgiev** [comment, marked as answer]
> Seems like [] operator can't be applied on the error object.

I don't understand what do you mean by this.

This is not PocketBase specific but to catch a specific error you can use either `is` if check inside the catch block or directly filter it before the catch:

```dart
try {
  ...
} on ClientException catch(e) {
  // e.response["data"]
}
```

In general for all available exception fields you can refer to [Error handling](https://github.com/pocketbase/dart-sdk?tab=readme-ov-file#error-handling) or the [class reference](https://pub.dev/documentation/pocketbase/latest/pocketbase/ClientException-class.html).

If you are still not able to resolve your issue, please paste the exact error.

**Comment by rodnnnney** [comment]
Thank you for the quick response! I just didn't know you had to specify `ClientExepction`  to access status codes and data etc.
